features/323-login-id/stage-1.md

```diff
@@ -0,0 +1,91 @@
+# Implict login ID key, Login ID realm
+
+## Database Schema
+
+- In `_auth_provider_password` table:
+    - A column `realm` (`TEXT NOT NULL`) would be added.
+    - The existing unique index on `login_id` would be changed to include `realm`.
+
+## Configuration
+
+- Add field `realmWhitelist` (list of strings, optional).
+
+## HTTP API
+- `POST /login`:
+    - Change type of `login_id` from map of strings to a plain string value.
+    - Add parameter `realm` (string, optional).
+    - Example:
+      ```json
+        {
+            "login_id": "test@example.com",
+            "password": "12345678"
+        }
+        {
+            "realm": "student",
+            "login_id": "test@example.com",
+            "password": "12345678"
+        }
+      ```
+- `POST /signup`
+    - Add parameter `realm` (string, optional).
+    - Example:
+      ```json
+        {
+            "realm": "teacher",
+            "login_ids": {
+                "email": "test@example.com"
+            },
+            "password": "12345678"
+        }
+        {
+            "realm": "student",
+            "login_id": {
+                "email": "test@example.com"
+            },
+            "password": "12345678"
+        }
+      ```
+
+- `POST /verify_request`
+    - Rename parameter `record_key` to `login_id_key`
+    - Add parameter `login_id` (string, optional).
+    - Example: 
+      ```json
+        {
+            "login_id_key": "email",
+            "login_id":"test@example.com"
+        }
+      ```
+
+
+## SDK API
+Following snippets use TypeScript:
+```diff
+- async signup(loginIDs: { [key: string]: string }, password: string, data?: unknown): Promise<User>;
+- async signupWithUsername(username: string, password: string, data?: unknown): Promise<User>;
+- async signupWithEmail(email: string, password: string, data?: unknown): Promise<User>;
```

Comment:
Consider don't remove `withUsername` and `withEmail` shortcut? Those seems make new Skygear users understand much quicker.

Reply:
It's just my personal opinion that it is redundant and not provide much convenience. I'll keep it in if you prefer.

Reply:
Yea should help new users to use it. It is because for new users, they would read the API docs quickly to understand what features are available, having the common features as Function Name instead of an parameter is helpful.